### PR TITLE
items: add new temporary_location field

### PR DESCRIPTION
**Pull request by BadrAly**
Adds a new field temporary_location for items to save Virtua
item location when item location is different than its parent.

The new field has a $ref to a location and an end_date. Currently
no logic is added to this implementation.

Co-Authored-by: Aly Badr <aly.badr@rero.ch>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json [hunk 1]
```diff
@@ -230,6 +231,55 @@
         }
       }
     },
+    "temporary_location": {
+      "title": "Temporary location",
+      "description": "Apply a temporary location for this item.",
```
Comment (iGormilhit) [inline rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json:236]:
```suggestion
      "description": "Set a temporary location for this item.",
```

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
items: add new temporary_location field

Adds a new `temporary_location` field to get only one holdings for
serial items actually in multiple locations.

The new field is a `$ref` to a location and has an `end_date`. Currently
no logic is added to this implementation.
```

**Review by iGormilhit — APPROVED** [review]

**Review by Garfield-fr — APPROVED** [review]

**Review by benerken — APPROVED** [review]

File: rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json [hunk 2]
```diff
@@ -230,6 +231,55 @@
...
+      "title": "Temporary location",
+      "description": "Set a temporary location for this item.",
+      "type": "object",
+      "propertiesOrder": [
+        "$ref",
+        "end_date"
+      ],
+      "required": [
+        "$ref"
+      ],
```
Comment (jma) [inline rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json:244]: additionnalProperty is missing

File: rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json [hunk 3]
```diff
@@ -230,6 +231,55 @@
         }
       }
     },
+    "temporary_location": {
+      "title": "Temporary location",
+      "description": "Set a temporary location for this item.",
```
Comment (jma) [inline rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json:236]: useless you can remove it

File: rero_ils/modules/items/mappings/v7/items/item-v0.0.1.json [hunk 4]
```diff
@@ -58,6 +58,19 @@
           }
         }
       },
+      "temporary_location": {
+        "properties": {
+          "type": {
```
Comment (jma) [inline rero_ils/modules/items/mappings/v7/items/item-v0.0.1.json:63]: seems not present in the JSONSchema

File: rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json [hunk 5]
```diff
@@ -230,6 +231,55 @@
...
+            "fieldMap": "location"
+          }
+        },
+        "end_date": {
+          "title": "Deletion date",
+          "type": "string",
+          "format": "date",
+          "pattern": "^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$",
+          "form": {
+            "hideExpression": "false",
```
Comment (jma) [inline rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json:263]: Why? Seems not effect, you can remove this.

**Review by jma — CHANGES_REQUESTED** [review]

**Review by jma — APPROVED** [review]

**Pull request merged**